### PR TITLE
Remove impossible condition

### LINKED ISSUE #4528: Auth token is not getting regenerated every time it is invalid
## Current behavior

If a customer token expires or is manually deleted from Magento 2 database, table **oauth_token**
upon next request, the token will be invalidated on VSF and recreated on Magento.

But, if it is deleted or it's expired once again, token invalidation and generation will **NOT** take place. An error will be shown asking the user to reload the page, after which a new token will be generated and everything will be working fine. 

## Expected behavior

The token should be generated without the need to reload the page.

## Steps to reproduce the issue

1. Log into frontend
2. Wait for the token to expire - you can also modify the creation time in the database so that you do not need to wait. **oauth_token** table. Or you can delete the token completely from the table.
3. Add a product (or do another action that requires auth token)
4. Delete or manually expire the newly generated token
5. Add a product
6. Error is shown asking to reload the page

## Can you handle fixing this bug by yourself?

- [X] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [X] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: Version 83.0.4103.97 (Official Build) (64-bit)
- OS: Linux Mint 19.3
- Node: v14.3.0
- Code Version: master

### PR DESCRIPTION
**Pull request by TerrorSquad**
### Related Issues

closes #4528

### Short Description and Why It's Useful
It will enable regeneration of auth token if that token is expired or deleted without the need to reload the page.

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [X] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [X] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: core/lib/sync/task.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ function _internalExecute (resolve, reject, task: Task, currentToken, currentCar
       if (responseCode !== 200) {
         if (responseCode === 401 /** unauthorized */ && currentToken) { // the token is no longer valid, try to invalidate it
           Logger.error('Invalid token - need to be revalidated' + currentToken + task.url + rootStore.state.userTokenInvalidateLock, 'sync')()
-          if (isNaN(rootStore.state.userTokenInvalidateAttemptsCount) || isUndefined(rootStore.state.userTokenInvalidateAttemptsCount)) rootStore.state.userTokenInvalidateAttemptsCount = 0
+          rootStore.state.userTokenInvalidateAttemptsCount = 0
           if (isNaN(rootStore.state.userTokenInvalidateLock) || isUndefined(rootStore.state.userTokenInvalidateLock)) rootStore.state.userTokenInvalidateLock = 0
```
Comment (gibkigonzo) [inline core/lib/sync/task.ts:78]: please remove those lines (77, 78). Invalidation should be fixed with those changes https://github.com/DivanteLtd/vue-storefront/pull/4882

**Review by gibkigonzo — CHANGES_REQUESTED** [review]
@TerrorSquad please check a comment and rebase to `hotfix/v1.12.3`

**Comment by TerrorSquad** [comment]
@gibkigonzo I updated the code. Can you review?

**Review by gibkigonzo — APPROVED** [review]

**Review by Fifciu — APPROVED** [review]

**Comment by TerrorSquad** [comment]
Guys, for some reason, the licence still says Pending. I accepted it multiple times, but nothing changed here. Any ideas?

**Pull request merged**